### PR TITLE
Shaders for field arithmetic

### PR DESCRIPTION
**Pull request by mitschabaude**
This PR adds initial shaders for addition, subtraction and multiplication on vectors of finite field elements
Also included is a simple TS script runnable with deno which validates the result against a simple bigint implementation and measures performance

### CODE REVIEW

File: wgsl/field5x4x13.wgsl [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    let zi4 = carry(z[4], zi3.carry);
+    z[4] = zi4.limb;
+    return Field(z);
+}
+
+
+// ========== main ==========
+
+// tasks per workgroup
+const tasksPerGroup: u32 = 1024u; // TODO how to declare an override?
```
Comment (fabrizio-m) [inline wgsl/field5x4x13.wgsl:182]: an uniform could also work for this, if that is what you are refereeing with override.

File: wgsl/field5x4x13.wgsl [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+@group(0)
+@binding(0)
+var<storage, read_write> xs: array<Field>;
+
+@group(0)
+@binding(1)
+var<storage, read> ys: array<Field>;
+
+@compute
+@workgroup_size(64)
```
Comment (fabrizio-m) [inline wgsl/field5x4x13.wgsl:193]: have you tried with `1`? I'm not exactly sure how it works, but my idea is that you use more than one when you want several threads to have shared storage.

File: wgsl/field5x4x13.wgsl (hunk 1 above)
Comment (mitschabaude) [reply wgsl/field5x4x13.wgsl:182]: supposedly there are three types of variables which get evaluated at different stages -- `const`, `override` and runtime (`let`, `var`). Override can be set from the outside

File: wgsl/field5x4x13.wgsl (hunk 2 above)
Comment (mitschabaude) [reply wgsl/field5x4x13.wgsl:193]: we should definitely try all combinations once we have a better GPU benchmark

**Pull request merged**